static/app/views/settings/projectIssueGrouping/upgradeGrouping.tsx

```diff
@@ -122,17 +146,17 @@ function UpgradeGrouping({
         >
           <Confirm
             disabled={noUpdates}
-            onConfirm={handleUpgrade}
-            priority={riskLevel >= 2 ? 'danger' : 'primary'}
-            confirmText={t('Upgrade')}
-            message={getModalMessage()}
+            onConfirm={handleConfirmUpgrade}
+            priority={priority}
+            confirmText={confirmText}
+            message={modalMessage}
           >
```

Comment:
I would suggest just changing your buttons `onClick` to `openUpgradeGroupingConfirmModal` and get rid of `Confirm` all together :)

Reply:
yes. Good idea.